### DISCUSSION TITLE
undefined reference to `z_impl_k_timer_status_get' , undefined reference to z_impl_k_timer_start'

### DISCUSSION THREAD

**Discussion by ghost** [zephyrproject-rtos/zephyr#65810] [Q&A]
Hello all,

I have implemented the timer functionality based on ref : (https://docs.zephyrproject.org/latest/doxygen/html/group__timer__apis.html#ga3ba70e9f059ff52fd2057ab89ea7f2ee)

while using the API's from kernel.h (k_timer_status_get, k_timer_start)  I'm getting this linker error  **undefined reference to `z_impl_k_timer_status_get' , undefined reference to `z_impl_k_timer_start'**  while compiling the project
do I miss something here ? or any additional config has to be done while using the timer API's ?
I'm using the following
Zephyr version : 3.3.99
Microcontroller : LPC55S16

can someone suggest a solution ... 

Thanks in Advance !!

**Comment by arbrauns** [comment]
Sounds like `kernel/timer.c` isn't being compiled for whatever reason. According to `kernel/CMakeLists.txt`, only `CONFIG_SYS_CLOCK_EXISTS` should be required, and it's enabled by default - check [`guiconfig`](https://docs.zephyrproject.org/latest/build/kconfig/menuconfig.html) to see if something has disabled it.
